nuxeo-core/nuxeo-core-io/src/main/java/org/nuxeo/ecm/core/io/marshallers/csv/DocumentPropertyCSVWriter.java

```diff
@@ -54,17 +63,38 @@ protected void writeHeader(Property entity, CSVPrinter printer) throws IOExcepti
 
     protected void writeScalarProperty(Property entity, CSVPrinter printer) throws IOException {
         Object value = entity.getValue();
+        Type type = entity.getType();
         if (value == null) {
             printer.print(null);
-        } else if (entity.getType() instanceof BinaryType) {
-            writeUnsupported(entity, printer);
+        } else if (type instanceof BinaryType) {
+            writeUnsupported(type, printer);
+        } else {
+            printer.print(type.encode(value));
+        }
+    }
+
+    protected void writeListProperty(Property entity, CSVPrinter printer) throws IOException {
+        ListType type = (ListType) entity.getType();
+        if (entity instanceof ArrayProperty) {
```

Comment:
do we need to support ListProperty as well ?

Reply:
I'd say yes at some point, but it's not the scope of this ticket